File: solutions/elixir/rna-transcription/1/lib/rna_transcription.ex
defmodule RnaTranscription do
  @dictionary %{ ?A => ?U,
                 ?C => ?G,
                 ?T => ?A,
                 ?G => ?C }

  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

    iex> RnaTranscription.to_rna(~c"ACTG")
    ~c"UGAC"
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    dna
    |> Enum.map(fn nucleotide -> @dictionary[nucleotide] end)
  end
end

File: solutions/javascript/rna-transcription/1/rna-transcription.js
export function toRna(dna) {
    var rna = "";
    [...dna].forEach((nucleotide) => {
        switch (nucleotide) {
            case 'G':
                rna += 'C';
                break;
            case 'C':
                rna += 'G';
                break;
            case 'T':
                rna += 'A';
                break;
            case 'A':
                rna += 'U';
                break;
        }
    })
    return rna;
};